### PR TITLE
[FIX] website_sale: base unit price not updated with pricelist and tax

### PR DESCRIPTION
**Pull request by thle-odoo**
Steps to reproduce:
- set a price on the product form;
- set a base unit count on the product form;
- define at least a pricelist with an other price;
- go to website on the product page;
- change the selected pricelist;
- change the Tax-Excluded/Included option.

Issue:
The base unit price ignores the tax included option (tax excluded/included).

Solution:
Calculate the base unit price using "list_price" (which takes into account the tax) when we need to display it without saving.

opw-2995474

### CODE REVIEW

**Comment by thle-odoo** [comment]
(@pierre-pde )
The fix was designed as follows:
The base unit price is a value only intended to be displayed to the customer (if I'm not mistaken).
The goal is to get the right value when calling the `_get_combination_info` function and thus be able to access it in the view via `combination_info['base_unit_price']`.
My opinion is that it is correct to use the `list_price` value that will be returned on the view to calculate the base unit price. Indeed, this last value of `list_price` takes into account the pricelist, the amount of the tax (if it is included) and the currency. This value is calculated in the `_get_combination_info` function.
It makes sense not to overwrite the starting base unit price with this new value (it will change all the time depending on the visitor).

@Williambraecky
You made a fix in 16.0 about this: https://github.com/odoo/odoo/pull/102543

That's why I would like your opinion on this fix.

File: addons/website_sale/views/templates.xml [hunk 1]
```diff
@@ -555,7 +555,7 @@
     </template>
 
     <template id="base_unit_price" name="Product Bas eunit price">
-        (<span class="o_base_unit_price" t-field="product.base_unit_price" t-options="{'display_currency': website.currency_id}"/>
+        (<span class="o_base_unit_price" t-esc="round(combination_info['base_unit_price'],2)"/> <span t-esc="website.currency_id.symbol"/>
```
Comment (Williambraecky) [inline addons/website_sale/views/templates.xml:558]: This removes compatibility with currencies which display the symbol on the left, it seems like we can use `t-out` along with `t-options-widget` instead.

File: addons/website_sale/views/templates.xml [hunk 2]
```diff
@@ -623,7 +623,7 @@
                                 <div class="js_product js_main_product mb-3">
                                     <div>
                                         <t t-call="website_sale.product_price" />
-                                        <small class="ml-1 text-muted o_base_unit_price_wrapper d-none" groups="website_sale.group_show_uom_price">
+                                        <small class="ml-1 text-muted o_base_unit_price_wrapper" groups="website_sale.group_show_uom_price">
```
Comment (Williambraecky) [inline addons/website_sale/views/templates.xml:626]: Why this change?

**Review by Williambraecky — CHANGES_REQUESTED** [review]
It seems ok 👍 thanks.
My fix seems to be incomplete/wrong but we were in a rush 😬

File: addons/website_sale/views/templates.xml (hunk 2 above)
Comment (thle-odoo) [reply addons/website_sale/views/templates.xml:626]: It's a class selector to prevent displaying the element.
```
.d-none {
  display: none !important;
}
```

File: addons/website_sale/views/templates.xml [hunk 3]
```diff
@@ -555,7 +555,7 @@
     </template>
 
     <template id="base_unit_price" name="Product Bas eunit price">
-        (<span class="o_base_unit_price" t-field="product.base_unit_price" t-options="{'display_currency': website.currency_id}"/>
+        (<span class="o_base_unit_price" t-esc="round(combination_info['base_unit_price'],2)" t-options="{'widget': 'monetary', 'display_currency': website.currency_id}"/>
```
Comment (Williambraecky) [inline addons/website_sale/views/templates.xml:558]: No need to round anymore, the widget will handle it for us.
There if a currency uses more than 2 digits we lose them on the widget.

File: addons/website_sale/views/templates.xml (hunk 2 above)
Comment (Williambraecky) [reply addons/website_sale/views/templates.xml:626]: I know what d-none does thanks 🤣 

I don't think this change is needed. That class is removed/readded by the JS code when necessary

**Review by Williambraecky — CHANGES_REQUESTED** [review]

**Review by Williambraecky — APPROVED** [review]
Thanks a lot, sorry for the chipoting

robodoo r+
